lib/logger/lib/logger/translator.ex

```diff
@@ -406,11 +406,9 @@ defmodule Logger.Translator do
   end
 
   defp report_crash(min_level, crashed, extra, linked) do
-    [
-      {:pid, pid},
-      {:registered_name, name},
-      {:error_info, {kind, reason, stack}} | crashed
-    ] = crashed
+    {pid, crashed} = Keyword.pop_first(crashed, :pid)
+    {name, crashed} = Keyword.pop_first(crashed, :registered_name)
+    {{kind, reason, stack}, crashed} = Keyword.pop_first(crashed, :error_info)
```

Comment:
I thought this should be semantically more robust to future changes, while still efficient enough popping from the head?

Reply:
Yeah this is totally fine, it's a really tiny list.